drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -275,24 +299,43 @@ int intel_adsp_hda_dma_stop(const struct device *dev, uint32_t channel)
 
 	intel_adsp_hda_disable(cfg->base, cfg->regblock_size, channel);
 
+#if CONFIG_PM_DEVICE
 	return pm_device_runtime_put(dev);
+#else
+	return 0;
+#endif
```

Comment:
This `#ifdef` guard is wrong, the right one is `CONFIG_PM_DEVICE_RUNTIME`, but as I mentioned you don't need it, just call it directly.

Reply:
I don't trust the compiler to remove all redundant logic if `CONFIG_PM_DEVICE_RUNTIME` is disabled.

Reply:
You don't have to trust anything ... If this option is disable the function is an inline function that just return 0. If you are worried about not inlining here, then we have a way bigger problem because a lot of more critical places are rely on it.
I am saying because that makes the code easier to maintain.